moto/cognitoidp/models.py

```diff
@@ -1643,13 +1645,39 @@ def set_user_mfa_preference(
                             "User has not verified software token mfa"
                         )
 
+                    if software_token_mfa_settings["PreferredMfa"]:
+                        user.preferred_mfa_setting = "SOFTWARE_TOKEN_MFA"
                 elif sms_mfa_settings["Enabled"]:
                     user.sms_mfa_enabled = True
 
+                    if sms_mfa_settings["PreferredMfa"]:
+                        user.preferred_mfa_setting = "SMS_MFA"
                 return None
         else:
             raise NotAuthorizedError(access_token)
 
+    def admin_set_user_mfa_preference(
+        self, user_pool_id, username, software_token_mfa_settings, sms_mfa_settings
+    ):
+        user = self.admin_get_user(user_pool_id, username)
+
+        if software_token_mfa_settings["Enabled"]:
```

Comment:
The docs do not say whether this is required, so it's probably safer to assume this field may not be set:
```
if software_token_mfa_settings.get("Enabled", False):
```